src/filesystem/translation.rs

```diff
@@ -21,9 +21,9 @@ impl Translator for FileSystem {
     /// the cwd.
     fn translate_path(&self, guest_path: &Path, deref_final: bool) -> Result<PathBuf> {
         let unchecked_guest_path = guest_path;
-        //TODO: dir_fd != AT_FDCWD
         let mut guest_path = if unchecked_guest_path.is_relative() {
             // It is relative to the current working directory.
+            warn!("translate_path() got a relative path, but it only accepts absolute paths now");
```

Comment:
How would we get a relative path? This needs a unit test.

Reply:
I am about to abandon this function, because it is basically a combination of two things: canonicalize() and substitute(). However sometimes we need to get intermediate results (a canonical path), so it would be better to use these two functions separately.